compiler-core/src/build/package_compiler.rs

```diff
@@ -63,7 +63,7 @@ pub struct PackageCompiler<'a, IO> {
     pub compile_beam_bytecode: bool,
     pub subprocess_stdio: Stdio,
     pub target_support: TargetSupport,
-    pub cached_warnings: CachedWarnings,
+    pub package_kind: PackageKind,
```

Comment:
The package compiler is a lower level API, not coupled to the project compiler, and is used in other contexts (wasm compiler, `gleam compiler-package`), so it cannot know about a ProjectCompiler concept such as the project compiler package kind. The behaviour needs to be specified specifically through flags such as CachedWarnings.